Homework 2

eogenio777 opened the pull request.

eogenio777 committed:
add rough  ex1

eogenio777 committed:
add ex2 tests, refactor ex1

akulebyakin commented on pom.xml:
```diff
@@ -4,13 +4,96 @@
+    <dependency>
+      <groupId>io.github.bonigarcia</groupId>
+      <artifactId>webdrivermanager</artifactId>
+      <version>5.1.0</version>
```
In some dependency versions you use property variables like `${testng.version}` but in some you use just version like `5.1.0`. Please choose one of the options.

eogenio777 replied:
All changed to variables.

akulebyakin commented on src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java:
```diff
@@ -0,0 +1,131 @@
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
```
Unused imports can be removed
```
import static org.assertj.core.api.Assertions.assertThat;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
```

eogenio777 replied:
All unused imports are removed.

akulebyakin commented on src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java:
```diff
@@ -0,0 +1,131 @@
+
+        // 8) Assert that there is the iframe with “Frame Button” exist
+        WebElement frame = driver.findElement(By.id("frame"));
+        softAssertions.assertThat(frame).isNotNull();
```
`driver.findElement` cannot return `null` because this method contains this check
```java
if (element == null) {
    throw new NoSuchElementException("Unable to find element with locator " + locator);
}
```
So I think this assertion `softAssertions.assertThat(frame).isNotNull();` is redundant here

eogenio777 replied:
Changed assertion to `softAssertions.assertThat(frame.isDisplayed()).isTrue();`.

akulebyakin commented on src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java:
```diff
@@ -0,0 +1,131 @@
+
+        driver.switchTo().frame("frame");
+        WebElement frameButton = driver.findElement(By.id("frame-button"));
+        softAssertions.assertThat(frameButton).isNotNull();
```
Since plain `findElement` cannot return `null` I suggest you replace this assertion by some other assertion, for example `element.isDisplayed()` and `element.isEnabled()`

eogenio777 replied:
Changed assertion to `softAssertions.assertThat(frameButton.isDisplayed()).isTrue();`

akulebyakin commented on src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java:
```diff
@@ -0,0 +1,131 @@
+        actualItems.forEach(element -> softAssertions.assertThat(element.isDisplayed()).isTrue());
+
+        // assert texts
+        List<WebElement> actualItemsTexts = new ArrayList<>(actualItems);
```
Looks like `actualItemsTexts` is just copying `actualItems`...

eogenio777 replied:
Changed `actualItemsTexts` type to String and now it contains only texts of the elements.

akulebyakin commented on src/test/java/com/epam/tc/hw2/ex2/Exercise2Test.java:
```diff
@@ -0,0 +1,124 @@
+            }
+        }
+        assertThat(dropdownNoticed).isTrue();
+
+
```
empty lines in the end of the method can be removed

eogenio777 replied:
Empty lines are removed

akulebyakin reviewed (changes requested):
Please review comments

eogenio777 committed:
Fix minor issues with style and logic

Change pom.xml versions from numbers to variables, remove unuesed imports,
change isNotNull assert to displayment check, change signature of actualItemsTexts:
new copying .getText() values (String), remove empty lines.

akulebyakin reviewed (approved):
Approved. Good job!

eogenio777 committed:
Merge branch 'master' into homework_2
